src/checks/kubernetes/services/ports/ports.go

```diff
@@ -38,39 +39,64 @@ type doesPortExistStruct struct {
 	} `yaml:"values"`
 }
 
-// DoesPortExist DoesPortExist
-func (i inputs) DoesPortExist() Results {
+func doesPortExistParse(valuesYaml string, v *doesPortExistStruct) error {
 
-	var values doesPortExistStruct
-
-	err := yaml.Unmarshal([]byte(i.valuesYaml), &values)
+	err := yaml.Unmarshal([]byte(valuesYaml), &v)
 	if err != nil {
 		log.Fatalf("Unmarshal: %v", err)
```

Comment:
```suggestion
		return errors.New("YAML Parse Error: "+err)
```

Reply:
This way a single checks invalid YAML won't break all the checks

Reply:
Good catch...i missed this.. Thanks!